### PR TITLE
Feature/input password

### PR DESCRIPTION
**Pull request by thomasowow**
image

Implements `InputPassword` component.

### CODE REVIEW

**Review by TimDirks — APPROVED** [review]

File: src/components/ButtonState/ButtonState.test.ts [hunk 1]
```diff
@@ -36,6 +36,24 @@ describe('template', () => {
...
+        const wrapper = shallowMount(ButtonState, {props: {disabled: true}});
+
+        expect(wrapper.vm.disabled).toBe(true);
+    });
+
+    it('is true when disabled and not loading', () => {
+        const wrapper = shallowMount(ButtonState, {props: {disabled: true, loading: false}});
+
+        expect(wrapper.vm.disabled).toBe(true);
+    });
```
Comment (albertpratomo) [inline src/components/ButtonState/ButtonState.test.ts:55]: Feels like this is already tested by the previous test above.
I think 'is false when disabled and loading' is a better test.

Comment (albertpratomo) [reply src/components/ButtonState/ButtonState.test.ts:55]: Wait, maybe it should also be disabled when loading haha

File: src/components/InputPassword/InputPassword.vue [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    xmlns="http://www.w3.org/2000/svg"
+                    class="input-password-icon"
+                    fill="currentColor"
+                    height="1em"
+                    viewBox="0 0 16 16"
+                >
+                    <path d="M13.359 11.238C15.06 9.72 16 8 16 8s-3-5.5-8-5.5a7.028 7.028 0 0 0-2.79.588l.77.771A5.944 5.944 0 0 1 8 3.5c2.12 0 3.879 1.168 5.168 2.457A13.134 13.134 0 0 1 14.828 8c-.058.087-.122.183-.195.288-.335.48-.83 1.12-1.465 1.755-.165.165-.337.328-.517.486l.708.709z" />
+                    <path d="M11.297 9.176a3.5 3.5 0 0 0-4.474-4.474l.823.823a2.5 2.5 0 0 1 2.829 2.829l.822.822zm-2.943 1.299.822.822a3.5 3.5 0 0 1-4.474-4.474l.823.823a2.5 2.5 0 0 0 2.829 2.829z" />
+                    <path d="M3.35 5.47c-.18.16-.353.322-.518.487A13.134 13.134 0 0 0 1.172 8l.195.288c.335.48.83 1.12 1.465 1.755C4.121 11.332 5.881 12.5 8 12.5c.716 0 1.39-.133 2.02-.36l.77.772A7.029 7.029 0 0 1 8 13.5C3 13.5 0 8 0 8s.939-1.721 2.641-3.238l.708.709zm10.296 8.884-12-12 .708-.708 12 12-.708.708z" />
+                </svg>
```
Comment (albertpratomo) [inline src/components/InputPassword/InputPassword.vue:32]: Should we separate this into other components? Feels more organized and more reusable for other things

**Review by albertpratomo — APPROVED** [review]

File: src/components/InputPassword/InputPassword.vue [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import FormControl from '@/components/FormControl';
+import InputGroup from '@/components/InputGroup';
+import {buttonProps} from '@/composables/useButton';
+import extractKeysFrom from '@/library/extractKeysFrom';
+import {formControlProps} from '@/composables/useFormControl';
+import {sizeProps} from '@/composables/useSize';
+
+export interface Rule {
+    feedback: string;
+    rule: RegExp;
```
Comment (barisbora) [inline src/components/InputPassword/InputPassword.vue:69]: I think good to have pass custom validation function for edge cases. For example can't be same as old password or checking same as password_confirm field..

Also I think should be async thus we can do some api call etc for validation.

```js
export interface Rule {
    feedback: string;
    rule: RegExp | (value: string) => Promise<boolean>;
    valid?: boolean;
}
```

**Review by barisbora — APPROVED** [review]

File: src/components/ButtonState/ButtonState.vue [hunk 4]
```diff
@@ -69,9 +70,10 @@ const props = defineProps({
         default: false,
     },
     successVariant: variantProp('success'),
+    ...disabledProps
```
Comment (PayamOWOW) [inline src/components/ButtonState/ButtonState.vue:73]:
```suggestion
    ...disabledProps,
```

**Review by PayamOWOW — APPROVED** [review]

**Comment by thomasowow** [comment]
Re-open if we still want this.

**Pull request closed without merging**